pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -66,7 +67,10 @@ func (t *Torch) Validate(_ client.Object, runtimeInfo *runtime.Info, _, newObj *
 		if numProcPerNode.Type == intstr.String {
 			allowed := sets.New("auto", "cpu", "gpu")
 			if !allowed.Has(numProcPerNode.StrVal) {
-				allErrs = append(allErrs, field.Invalid(numProcPerNodePath, newObj.Spec.Trainer.NumProcPerNode, fmt.Sprintf("must have an int value or %v", allowed.UnsortedList())))
+				allowedList := sets.New("auto", "cpu", "gpu").UnsortedList()
+				// Make sure the order is decided for UT
+				sort.Strings(allowedList)
+				allErrs = append(allErrs, field.Invalid(numProcPerNodePath, numProcPerNode, fmt.Sprintf("must have an int value or %v", allowedList)))
```

Comment:
```suggestion
				allErrs = append(allErrs, field.Invalid(numProcPerNodePath, numProcPerNode, fmt.Sprintf("must have an int value or %v", sets.List(allowed))))
```

We could leveraging `sets.List` function.

Reply:
Got, I have updated with `sets.List`.